indexing-service/src/test/java/org/apache/druid/indexing/common/task/batch/parallel/Factory.java

```diff
@@ -0,0 +1,273 @@
+import com.google.common.collect.ImmutableMap;
+import org.apache.druid.client.indexing.IndexingServiceClient;
+import org.apache.druid.data.input.impl.DimensionsSpec;
+import org.apache.druid.data.input.impl.JSONParseSpec;
+import org.apache.druid.data.input.impl.StringInputRowParser;
+import org.apache.druid.data.input.impl.TimestampSpec;
+import org.apache.druid.guice.FirehoseModule;
+import org.apache.druid.indexer.partitions.HashedPartitionsSpec;
+import org.apache.druid.indexer.partitions.PartitionsSpec;
+import org.apache.druid.indexer.partitions.SingleDimensionPartitionsSpec;
+import org.apache.druid.indexing.common.task.IndexTaskClientFactory;
+import org.apache.druid.indexing.common.task.TaskResource;
+import org.apache.druid.jackson.JacksonModule;
+import org.apache.druid.java.util.common.Intervals;
+import org.apache.druid.java.util.common.granularity.Granularities;
+import org.apache.druid.java.util.common.jackson.JacksonUtils;
+import org.apache.druid.java.util.http.client.HttpClient;
+import org.apache.druid.segment.TestHelper;
+import org.apache.druid.segment.indexing.DataSchema;
+import org.apache.druid.segment.indexing.granularity.ArbitraryGranularitySpec;
+import org.apache.druid.segment.indexing.granularity.GranularitySpec;
+import org.apache.druid.segment.realtime.appenderator.AppenderatorsManager;
+import org.apache.druid.segment.realtime.firehose.InlineFirehoseFactory;
+import org.apache.druid.timeline.partition.HashBasedNumberedShardSpec;
+import org.easymock.EasyMock;
+import org.joda.time.Duration;
+import org.joda.time.Interval;
+
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+class Factory
+{
+  static final String AUTOMATIC_ID = null;
+  static final String ID = "id";
+  static final String GROUP_ID = "group-id";
+  static final TaskResource TASK_RESOURCE = null;
+  static final String SUPERVISOR_TASK_ID = "supervisor-task-id";
+  static final int NUM_ATTEMPTS = 1;
+  static final Map<String, Object> CONTEXT = Collections.emptyMap();
+  static final IndexingServiceClient INDEXING_SERVICE_CLIENT = null;
+  static final IndexTaskClientFactory<ParallelIndexSupervisorTaskClient> TASK_CLIENT_FACTORY = null;
+  static final AppenderatorsManager APPENDERATORS_MANAGER = null;
+  static final HttpClient SHUFFLE_CLIENT = null;
+  static final List<Interval> INPUT_INTERVALS = Collections.singletonList(Intervals.ETERNITY);
+  static final String TASK_EXECUTOR_HOST = "task-executor-host";
+  static final int TASK_EXECUTOR_PORT = 1;
+  static final boolean USE_HTTPS = true;
+  static final Interval INTERVAL = Intervals.ETERNITY;
+  static final int NUM_ROWS = 2;
+  static final long SIZE_BYTES = 3;
+  static final int PARTITION_ID = 4;
+  static final String HOST = "host";
+  static final int PORT = 1;
+  static final String SUBTASK_ID = "subtask-id";
+  private static final ObjectMapper NESTED_OBJECT_MAPPER = TestHelper.makeJsonMapper();
```

Comment:
Hmm, maybe `TestUtils.getTestObjectMapper()` is more useful. It's lame that we have a couple of helper classes to make an `ObjectMapper` though.

Reply:
What's the advantage of using the object mapper from TestUtils versus TestHelper. Both are probably more than what's actually needed for the current usages of NESTED_OBJECT_MAPPER.